### PR TITLE
feat: retrieval stats

### PR DESCRIPTION
**Pull request by jcace**
Adds an interface for autoretrieve to POST retrieval stats data into the storage market db

### CODE REVIEW

**Review by jimmylee — APPROVED** [review]
LGTM, awesome thanks @jcace

File: README.md [hunk 1]
```diff
@@ -12,9 +12,22 @@ DOCUMENT_DATABASE_PORT=5432
 DOCUMENT_DATABASE_PASSWORD=xxxx
 ```
 
+### Running a local Development Database
```
Comment (jcace) [inline README.md:15]: @jimmylee lmk how you've been testing locally. I made it super easy to run a local `postgres` in docker here for testing new stuff.

File: scripts/create.js [hunk 2]
```diff
@@ -4,8 +4,8 @@ const NAME = `create.js`;
 
 console.log(`RUNNING: ${NAME}`);
 
-const createRun = DB.schema.createTable('filecoin_storage_providers', function (table) {
-  table.uuid('id').primary().unique().notNullable().defaultTo(DB.raw('uuid_generate_v4()'));
+const createFilecoinStorageProvidersTable = DB.schema.createTable('filecoin_storage_providers', function (table) {
+  table.uuid('id').primary().unique().notNullable().defaultTo(DB.raw('gen_random_uuid()'));
```
Comment (jcace) [inline scripts/create.js:8]: @jimmylee what version of postgres are we using in prod? the `uuid_generate_v4()` function wasn't playing nice with the docker image as it required the `uuid` extension to be installed, but it's builtin as `gen_random_uuid()` for v13 and onwards.

Comment (jimmylee) [reply scripts/create.js:8]: We're using 14. you can fix this by

`Promise.all([db.raw('CREATE EXTENSION IF NOT EXISTS "uuid-ossp"')]);`

And I would just add a setup script that looks like

```
import configs from "../knexfile";
import knex from "knex";

const name = `database-setup.js`;
const environment =
  process.env.NODE_ENV !== "production" ? "development" : "production";
const envConfig = configs[environment];
const db = knex(envConfig);

console.log(`RUNNING: ${name} NODE_ENV=${environment}`);

Promise.all([db.raw('CREATE EXTENSION IF NOT EXISTS "uuid-ossp"')]);

console.log(`FINISHED: ${name} NODE_ENV=${environment} (⌘ + C to quit)`);
```

Into the codebase

**Review by jimmylee — APPROVED** [review]
I have just been doing development off of production because everything is still in development, having a local database doesn't hurt though, obviously we need one :-)

I don't really use docker for that, I tend to just run the right postgres version locally and connect to it

This looks the same, We are using Postgres 14

**Pull request merged**